packages/template-retail-react-app/app/commerce-api/einstein.js

```diff
@@ -90,6 +94,105 @@ class EinsteinAPI {
         return this.einsteinFetch(endpoint, method, body)
     }
 
+    /**
+     * Tells the Einstein engine when a user views search results.
+     **/
+    async sendViewSearch(searchText, searchResults, args) {
+        const endpoint = `/activities/${this.config.siteId}/viewSearch`
+        const method = 'POST'
+
+        const products = searchResults.hits.map((product) => {
+            const {productId, sku = '', altId = '', altIdType = ''} = product
+            return {
+                id: productId,
+                sku,
+                altId,
+                altIdType
+            }
+        })
+
+        const body = {
+            searchText,
+            products,
+            ...args
+        }
+
+        return this.einsteinFetch(endpoint, method, body)
+    }
```

Comment:
No action required on this comment. But I wanted to make sure that you notice the commonalities between all these 'send' functions. There is probably room to DRY things up, but I don't expect this to be done at the moment as this Einstein work at some point will be removed from the template. It would be nice to hear your thoughts on what we might do in the future to make things more upgradable.